samples/file-encryptor/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+cmake_minimum_required(VERSION 3.10)
+
+project("File Encryptor Sample" LANGUAGES C CXX)
+
+find_package(OpenEnclave CONFIG REQUIRED)
+
+set(CMAKE_CXX_STANDARD 11)
+
+add_subdirectory(enc)
+add_subdirectory(host)
+
+# Generate key
+add_custom_command(OUTPUT private.pem public.pem
+  COMMAND openssl genrsa -out private.pem -3 3072
+  COMMAND openssl rsa -in private.pem -pubout -out public.pem)
+
+# Sign enclave
+add_custom_command(OUTPUT enc/enclave.signed
+  DEPENDS ${CMAKE_BINARY_DIR}/private.pem
+  COMMAND openenclave::oesign sign $<TARGET_FILE:enclave> ${CMAKE_SOURCE_DIR}/enc/file-encryptor.conf private.pem)
+
+if ((NOT DEFINED ENV{OE_SIMULATION}) OR (NOT $ENV{OE_SIMULATION}))
```

Comment:
Wait, this confuses me more. Can we just take `OE_SIMULATION` as `-DOE_SIMULATION=ON` (or `OFF`) when we call the script, and then check `if (OE_SIMULATION)` here; I really don't like adding an extra vector to send information to this script.

Reply:
That would be ideal, but we have to keep in mind that the [Docs](https://github.com/Microsoft/openenclave/blob/e0d3ce41926287d418ba44f403c651c045fcc76b/docs/GettingStartedDocs/SimulatorGettingStarted.md#run-unittests) and [code](https://github.com/Microsoft/openenclave/blob/710702db7ca1a0bfdd45deb53c74f5c7a784deff/host/tests.c#L15) are using it as an environment variable. Creating a special case with `-DOE_SIMULATION=ON/OFF` just for the samples would add more confusion in my opinion.
We can change the usage to a `-D` as whole but not part of this effort.

Reply:
Fair enough!

Reply:
Filed #1587.